### PR TITLE
Added lazy Loading

### PR DESCRIPTION
**Pull request by devDEV03**
By loading only the necessary resources for the initial view, lazy loading reduces the page's initial load time, which leads to faster rendering and a smoother experience for the user. This is especially important for users on slower networks or devices.

Lazy loading avoids loading resources until they are actually needed, saving bandwidth for both the server and the client. This is particularly beneficial for mobile users who may have limited data plans or slower connections.

By reducing the number of resources loaded initially, lazy loading helps minimize CPU and memory usage, which can improve the performance of devices with limited resources, such as smartphones and tablets.

### CODE REVIEW

File: src/components/Body.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+const StarredMovies = lazy(() => import("./StarredMovies"));
+const Shimmer = lazy(() => import("./Shimmer"));
+
+
 const Body = () => {
   const AppRouter = createBrowserRouter([
     {
       path: "/",
-      element: <Login />,
+      element: <Suspense fallback={<Shimmer />}><Login /></Suspense>,
```
Comment (mohitahlawat2001) [inline src/components/Body.js]: you can make lazy loading to all pages 
it need to be added only , pages which are not used frequently

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]
image
getting this error

File: src/components/Body.js (hunk 1 above)
Comment (devDEV03) [reply src/components/Body.js]: okay making the required changes

**Comment by devDEV03** [comment]
kindly check now
actually i think the problem was in MovieShimmer which i was importing by default while it was a named export, if u want i can create another shimmer ui

**Comment by mohitahlawat2001** [comment]
yes need to add different shimmer for every page , you lazy loaded ,  according to there ui

**Comment by devDEV03** [comment]
i can create a different pull request for every page. for now i can add just one default shimmer instead of shimmer ui for every page.What you say?

**Comment by mohitahlawat2001** [comment]
image
getting error while building 
also post photo of every shimmer you make

**Comment by devDEV03** [comment]
Okay,dw i will complete it by today

**Comment by devDEV03** [comment]
created the shimmers and configured them and also built it
browseShimmer
starredShimmer
loginShimmer

File: src/components/Body.js [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
-import Login from "./Login";
-import Movie from "./Movie";
-import StarredMovies from "./StarredMovies";
+import { lazy, Suspense } from "react";
+import { MovieShimmer } from "./Shimmer";
+import LoginShimmer from "./LoginShimmer";
+import StarredShimmer from "./StarredShimmer";
+import BrowserShimmer from "./BrowserShimmer";
+
+const Login  = lazy(() => import("./Login"));
```
Comment (mohitahlawat2001) [inline src/components/Body.js]: you can not lazy load first page i mention in last comment also

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]

File: src/components/StarredShimmer.js
Comment (mohitahlawat2001) [inline src/components/StarredShimmer.js:1]: need to improve this shimmer, not up standard i am looking for

File: src/components/HeaderShimmer.js
Comment (mohitahlawat2001) [inline src/components/HeaderShimmer.js:1]: this does not need shimmer

File: src/components/LoginShimmer.js
Comment (mohitahlawat2001) [inline src/components/LoginShimmer.js:1]: you can remove this also

**Review by mohitahlawat2001 — CHANGES_REQUESTED** [review]

File: src/components/HeaderShimmer.js
Comment (devDEV03) [reply src/components/HeaderShimmer.js:1]: Its a part of the other shimmers(Modularity
)

File: src/components/LoginShimmer.js
Comment (devDEV03) [reply src/components/LoginShimmer.js:1]: Okay

File: src/components/StarredShimmer.js
Comment (devDEV03) [reply src/components/StarredShimmer.js:1]: Okay lemme look into it

**Pull request closed without merging**